test/torchaudio_unittest/functional/functional_cpu_test.py

```diff
@@ -70,46 +70,124 @@ def test_pitch(self, frequency):
-        multiplier = 10.
-        power = 1.
-
-        db = F.amplitude_to_DB(x, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, torch.abs(x), atol=5e-5, rtol=1e-5)
-
         # Spectrogram power -> DB -> power
-        db = F.amplitude_to_DB(spec, multiplier, amin, db_multiplier, top_db=None)
-        x2 = F.DB_to_amplitude(db, ref, power)
-
-        torch.testing.assert_allclose(x2, spec, atol=5e-5, rtol=1e-5)
+        db = F.amplitude_to_DB(spec, POWER_MULT, AMIN, DB_MULT, top_db=None)
+        x2 = F.DB_to_amplitude(db, REF, 1.)
+
+        self.assertEqual(x2, spec)
+
+    def test_amplitude_to_DB_batch(self):
+        spec = torch.rand([2, 2, 100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def test_amplitude_to_DB_channels(self):
+        spec = torch.rand([2, 100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def test_amplitude_to_DB_freq(self):
+        spec = torch.rand([100, 100]) * 200
+        self._ensure_reversible(spec)
+
+    def _check_top_db(self, spec):
+        AMPLITUDE_MULT = 20.
+        AMIN = 1e-10
+        REF = 1.0
+        DB_MULT = math.log10(max(AMIN, REF))
+        TOP_DB = 40.
+
+        decibels = F.amplitude_to_DB(spec, AMPLITUDE_MULT, AMIN,
+                                     DB_MULT, top_db=TOP_DB)
+        above_top = decibels >= 6.0205
+        assert above_top.all(), decibels
+
+    def test_top_db_batch(self):
+        spec = torch.rand([1, 2, 100, 100]) * 200
+        # Predictability
+        spec[0, 0, 1] = 0
+        spec[0, 0, 0] = 200
+        self._check_top_db(spec)
+
+    def test_top_db_channel(self):
+        spec = torch.rand([1, 2, 100, 100]) * 200
+        # Predictability
+        spec[0, 0, 1] = 0
+        spec[0, 0, 0] = 200
+        self._check_top_db(spec[0])
+
+    def test_top_db_freq(self):
+        spec = torch.rand([1, 2, 100, 100]) * 200
+        # Predictability
+        spec[0, 0, 1] = 0
+        spec[0, 0, 0] = 200
+        self._check_top_db(spec[0][0])
+
+    def test_batched(self):
+        AMPLITUDE_MULT = 20.
+        AMIN = 1e-10
+        REF = 1.0
+        DB_MULT = math.log10(max(AMIN, REF))
+        TOP_DB = 40.
+
+        # Make a batch of noise
+        spec = torch.rand([2, 2, 100, 100]) * 200
+        # Make the second item blow out the first
+        spec[0] *= 0.5
+        # Predictability
+        spec[0, 0, 0, 0] = 0
+        spec[1, 0, 0, 0] = 200
+
+        # Ensure separate clamps are applied per-item in a batch.
+        batchwise_dbs = F.amplitude_to_DB(spec, AMPLITUDE_MULT, AMIN,
+                                          DB_MULT, top_db=TOP_DB)
+        itemwise_dbs = torch.stack([
+            F.amplitude_to_DB(item, AMPLITUDE_MULT, AMIN,
+                              DB_MULT, top_db=TOP_DB)
+            for item in spec
+        ])
+
+        self.assertEqual(batchwise_dbs, itemwise_dbs)
+
+    def test_per_spectrogram(self):
```

Comment:
I think this test is the fundamental response to #994. Can you refer that in docstring?

Example;

https://github.com/pytorch/audio/blob/422edb183f25ab91eb967381d328384e48c7a4a0/test/torchaudio_unittest/sox_io_backend/load_test.py#L359-L368